src/consensus/block_producer.rs

```diff
@@ -212,9 +212,18 @@ where
             let is_last = slice_index.is_max() || maybe_duration.is_none();
             if is_last && !parent_ready_receiver.is_terminated() {
```

Comment:
This is only to handle the case where we produce all slices before seeing the ParentReady, right?